opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -0,0 +1,1064 @@
+    const void *seg0_data = (const void *)((uintptr_t) segments[0].seg_addr.pval + skip_bytes);
+    size_t seg0_len = segments[0].seg_len - skip_bytes;
+
+    if (seg0_len > 0) {
+        BTL_VERBOSE(("unpacking %" PRIsize_t " bytes from segment 0 to %p", seg0_len, (void *) addr));
+        memcpy ((void *) addr, seg0_data, seg0_len);
+        addr += seg0_len;
+    }
+
+    for (size_t i = 1 ; i < segment_count ; ++i) {
+        size_t seg_len = segments[i].seg_len;
+        BTL_VERBOSE(("unpacking %" PRIsize_t " bytes from segment %" PRIsize_t " to %p", seg_len,
+                     i, (void *) addr));
+        memcpy ((void *) addr, segments[i].seg_addr.pval, seg_len);
+        addr += seg_len;
+    }
+}
+
+/**
+ * @brief copy data from a local address into a segment
+ *
+ * @in addr          location to read data from
+ * @in skip_bytes    bytes of segment data to skip before copying
+ * @in segments      segments to copy data to
+ * @in segment_count number of segments
+ */
+static void mca_btl_base_copy_to_segments (uint64_t addr, size_t max_len, size_t skip_bytes,
+                                           mca_btl_base_segment_t *segments,
+                                           size_t segment_count) {
+    void *seg0_data = (void *)((uintptr_t) segments[0].seg_addr.pval + skip_bytes);
+    size_t seg0_len = size_t_min (max_len, segments[0].seg_len - skip_bytes);
+
+    if (seg0_len > 0) {
+        BTL_VERBOSE(("packing %" PRIsize_t " bytes from 0x%" PRIx64 " to segment 0",
+                     seg0_len, addr));
+        memcpy (seg0_data, (const void *) addr, seg0_len);
+        addr += seg0_len;
+        max_len -= seg0_len;
+        segments[0].seg_len = seg0_len + skip_bytes;
+    }
+
+    for (size_t i = 1 ; i < segment_count && max_len ; ++i) {
+        size_t seg_len = size_t_min (segments[i].seg_len, max_len);
+
+        BTL_VERBOSE(("packing %" PRIsize_t " bytes from 0x%" PRIx64 " to segment %" PRIsize_t,
+                     seg_len, addr, i));
+
+        memcpy (segments[i].seg_addr.pval, (const void *) addr, seg_len);
+        segments[i].seg_len = seg_len;
+
+        addr += seg_len;
+        max_len -= seg_len;
+    }
+}
+
+static void mca_btl_base_am_queue_initiator_descriptor (mca_btl_base_module_t *btl,
+                                                        struct mca_btl_base_endpoint_t *endpoint,
+                                                        mca_btl_base_descriptor_t *descriptor)
+{
+    mca_btl_base_am_rdma_queued_descriptor_t *queued_descriptor =
+        OBJ_NEW(mca_btl_base_am_rdma_queued_descriptor_t);
+
+    queued_descriptor->btl = btl;
+    queued_descriptor->endpoint = endpoint;
+    queued_descriptor->descriptor = descriptor;
+
+    OPAL_THREAD_SCOPED_LOCK(&default_module.mutex,
+                            opal_list_append (&default_module.queued_initiator_descriptors,
+                                              &queued_descriptor->super));
+}
+
+static inline
```

Comment:
your function declaration format inconsistencies make my OCD hurt...

Reply:
C&P from multiple sources with different formatting. Get too used to ```p4 fix```. I created a clang-format style file and ran the new files through it. Should be completely consistent now. Will open a PR for adding the clang-format file to contrib.